### ISSUE TITLE
Problem with updating collection of associated resources

### ISSUE THREAD

**Issue by kmochocki** [spring-projects/spring-data-rest#2394]
Hi,

In our project we have a few resources, and one of them is aggregating the rest with the OneToMany relations (per each resource). During the creation (POST) of the main resource we're providing the URIs to the associated resources and eveything works fine (providing zero, one or more links). 

The issues apear when we try to update the main resource. While using PATCH, we can modify the collections with associations (add, reduce, remove all), but when we're using PUT, no changes are propagated. 

We proceeded with using PUT as the only way to update the resources and prior to 4.3.X version we had a really hacky way to achieve the expected result (using a similar request body structure for POST and PUT which updates the collections), but now it doesn't work.

Please have a look on [the sample project](https://github.com/kmochocki/data-rest-example) and let us know if the setup or assumptions are wrong.

**Comment by ruthst00** [comment]
#### This is a real bug which lives in the interaction between two key components:

#### 1. `DomainObjectReader.mergeForPut()` — the PUT merge path

When a PUT request arrives, `PersistentEntityResourceHandlerMethodArgumentResolver.readPutForUpdate()` calls `JsonPatchHandler.applyPut()` → `DomainObjectReader.readPut()` → `DomainObjectReader.mergeForPut()`.

Inside `mergeForPut()`, a `MergingPropertyHandler` is applied to all properties, and then a `LinkedAssociationSkippingAssociationHandler` is applied to all associations:

```java
it.doWithProperties(propertyHandler);
it.doWithAssociations(new LinkedAssociationSkippingAssociationHandler(associationLinks, propertyHandler));
```

#### 2. `LinkedAssociationSkippingAssociationHandler` — the culprit

```java
@Override
public void doWithAssociation(Association<? extends PersistentProperty<?>> association) {
    if (associations.isLinkableAssociation(association)) {
        return;  // ← SKIPS the association entirely
    }
    delegate.doWithPersistentProperty(association.getInverse());
}
```

If the association (e.g., a `@OneToMany` to another exported repository resource) is a **linkable association** (i.e., it has its own REST endpoint like `/api/mainResource/{id}/items`), it is **completely skipped** during the PUT merge. The property value from the incoming payload is never copied to the target object.

#### 3. The same skip exists in `doMerge()` (the PATCH/merge-patch path)

```java
if (!rawValue.isPresent() || associationLinks.isLinkableAssociation(property)) {
    continue;  // ← also skips linkable associations
}
```

So **both** PUT and merge-PATCH skip linkable associations when processing the entity body. The reason PATCH *appears* to work is that Spring Data REST also supports PATCH directly on the association sub-resource (e.g., `PATCH /api/mainResource/{id}/items`) via `RepositoryPropertyReferenceController`, which is a completely separate code path. When users say "PATCH works," they are likely using that sub-resource endpoint, not patching the main resource body.

---

### Why this design exists (and why it's still a bug)

The skip was intentional: linkable associations are meant to be managed via their own sub-resource URIs (e.g., `PUT /api/mainResource/{id}/items`). The body of a PUT to the main resource is not supposed to carry raw embedded objects for those associations.

**However**, the issue reporter's use case is valid: they want to send URI links (not embedded objects) in the PUT body to update the association collection — the same way POST accepts them. The framework accepts URI links in POST bodies for associations (handled by `PersistentEntityJackson2Module`'s deserializer), but the PUT merge path in `DomainObjectReader.mergeForPut()` / `LinkedAssociationSkippingAssociationHandler` discards those already-deserialized association values before they ever reach the save call.

In short: **the deserialized association values from the PUT body are silently thrown away**, making it impossible to update a collection of associated resources via a PUT to the parent resource body, even when providing valid URI links — which is inconsistent with how POST behaves.

I'm happy to submit a PR to fix this issue if everyone agrees with the assessment above.
